pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -304,6 +307,9 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
             elif self.trainer.amp_backend == AMPType.APEX:
                 checkpoint['amp_scaling_state'] = amp.state_dict()
 
+            # dump logger
+            checkpoint['logger'] = self.trainer.logger
```

Comment:
I don't think we should pickle the logger as it will break the checkpoint if their API changes.
Better to save their states instead.

Reply:
I was originally saving the states but it looks like a number of the loggers have actually taken steps to make sure they are picklable and that they restore correctly (CometLogger being one of them) although I'm not sure what that was done because as far as I can tell there wasn't any saving of loggers happening. 

I like this better because it's simpler and it matches up with the rest of the saving code which seems to prefer pickling objects over saving and restoring states. This way the logic of making object serializable is the responsibility of the person who writes the object (by making it picklable), the internal state of the logger is often not easy to serialize generically, so it's either this or we add a function to the logger API for returning and restoring state that every logger then has to implement. 

Also it wouldn't break the checkpoint because I allow logger loading to fail without disrupting the rest of the resume process. Probably we should also allow logger saving to fail in case a logger isn't picklable.

Reply:
I took another look at this and decided to go with the state saving API, I'm still working out why some of the tests are failing but please leave your feedback while im doing that. I modeled the API loosely after the callback state saving API